notebooks/MIRI_LRS_spectral_extraction/miri_lrs_advanced_extraction_part1.ipynb

```diff
@@ -0,0 +1,904 @@
+{
```

Comment:
<p>Considering the scope of the <code>json_ref_default</code> variable, you can combine both blocks using a context manager for file handling. One way to structure it would be :</p><pre class="ql-syntax" spellcheck="false">file_path = 'data/jw02072-o001_t010_miri_p750l_x1d_1089.fits'
with fits.open(file_path) as hdul:
&nbsp; &nbsp; header = hdu[0].header
&nbsp; &nbsp; json_ref_default = crds.getreferences(header)['extract1d']
&nbsp; &nbsp;&nbsp;
&nbsp; &nbsp; with open(json_ref_default) as json_ref:
&nbsp; &nbsp; &nbsp; &nbsp; x1dref_default = json.load(json_ref)
&nbsp; &nbsp; &nbsp; &nbsp; print('Settings for SLIT data: {}'.format(x1dref_default['apertures'][0]))
&nbsp; &nbsp; &nbsp; &nbsp; print('&nbsp; ')
&nbsp; &nbsp; &nbsp; &nbsp; print('Settings for SLITLESS data: {}'.format(x1dref_default['apertures'][1])) 
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/spacetelescope/jdat_notebooks/pull/93/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>